### PR TITLE
feat(backend): stop compressing artifacts

### PR DESCRIPTION
**Pull request by Ark-kun**
DIsabled compressing artifacts to .tar.gz archives.
Fixes https://github.com/kubeflow/pipelines/issues/1383

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/1640)

### CODE REVIEW

**Comment by Ark-kun** [comment]
@rileyjbauer Can you please check that UI and visualizations work fine when the artifacts are not compressed?

**Comment by Ark-kun** [comment]
@rileyjbauer I find it surprising that e2e frontend tests do not fail change. Can you please take a look?
@gaoning777  I find it surprising that sample tests do not fail after this change. Can you please take a look?

**Comment by Ark-kun** [comment]
I've changed the PR to only stop compressing the non-metadata artifacts. So this should be safe.

**Comment by Bobgy** [comment]
After you stopped compressing those artifacts, is there any code that could break?

I'm not very clear on where these artifacts are used.

**Comment by Ark-kun** [comment]
> After you stopped compressing those artifacts, is there any code that could break?
> 
> I'm not very clear on where these artifacts are used.

The UX that shows the artifacts can break with high probability.

@IronPan WDYT? How does the backend API handler compressed artifact blobs? What about directories?

**Comment by StefanoFioravanzo** [comment]
Any update on this PR? It would be very useful!

**Comment by StefanoFioravanzo** [comment]
Any update on this PR? It would be very useful!

**Comment by Ark-kun** [comment]
I think one problem right now is that some APIs like ReadArtifact probably only handle single files and will fail with directories.

**Comment by Bobgy** [comment]
This is sth I'd want to try sooner or later

File: backend/src/apiserver/resource/resource_manager.go [hunk 1]
```diff
@@ -218,6 +218,8 @@ func (r *ResourceManager) CreateRun(apiRun *api.Run) (*model.RunDetail, error) {
 		for artIdx, artifact := range template.Outputs.Artifacts {
 			if artifact.Name == "mlpipeline-ui-metadata" || artifact.Name == "mlpipeline-metrics" {
 				workflow.Workflow.Spec.Templates[templateIdx].Outputs.Artifacts[artIdx].Optional = true
+			} else {
+				artifact.Archive = &workflowapi.ArchiveStrategy{None: &workflowapi.NoneStrategy{}}
```
Comment (Bobgy) [inline backend/src/apiserver/resource/resource_manager.go:222]: nit: why do we add this in an else clause here?
I think we can add it to all.

**Review by Bobgy — COMMENTED** [review]
I think this is a better onboarding experience, also UX should already support this thanks to @eterna2's previous improvements.

I'd still want to make sure archive is an option, because text files can have a large compress rate, so it saves network IO. Can we add this as an option to API server config?

File: backend/src/apiserver/resource/resource_manager.go (hunk 1 above)
Comment (Bobgy) [reply backend/src/apiserver/resource/resource_manager.go:222]: Another issue is that, there should be some similar code in `CreateJob` method, you should add this part there too

Comment (Ark-kun) [reply backend/src/apiserver/resource/resource_manager.go:222]: >nit: why do we add this in an else clause here?

I'm not sure that Frontend and Backend Metrics are prepared for uncompressed artifacts.

Comment (eterna2) [reply backend/src/apiserver/resource/resource_manager.go:222]: Frontend server should support both compressed and uncompressed artifacts. 

https://github.com/kubeflow/pipelines/blob/4e5d724fef23d504ff1be02191c0f992672dff37/frontend/server/minio-helper.ts#L134

**Comment by eterna2** [comment]
This PR should also resolve #5558

**Comment by Bobgy** [comment]
@eterna2 in v2 compatible mode, artifacts are already not compressed by default

**Comment by eterna2** [comment]
Nice! I was reading thru the codes for v2. Am very excited with all the new stuff coming up!

**Comment by rimolive** [comment]
Closing this PR. No activity for more than a year.

/close

**Pull request closed without merging**